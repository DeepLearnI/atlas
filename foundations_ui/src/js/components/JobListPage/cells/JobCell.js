import React, { Component } from 'react';
import PropTypes from 'prop-types';

class JobCell extends Component {
  constructor(props) {
    super(props);
    this.state = {
      value: this.props.value,
    };
  }

  render() {
    const { value } = this.state;

    return (
      <p className="job-cell">{value}</p>
    );
  }
}

JobCell.propTypes = {
  value: PropTypes.any,
};

JobCell.defaultProps = {
  value: '',
};

export default JobCell;
