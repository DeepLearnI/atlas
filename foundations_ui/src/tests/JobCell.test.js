import React from 'react';
import ReactDOM from 'react-dom';
import JobCell from '../js/components/JobListPage/cells/JobCell';
import { shallow } from 'enzyme';
import configureTests from './setupTests';

configureTests();

it('Shallow Renders Job Cell', () => {
  const wrapper = shallow(<JobCell/>);
  expect(wrapper).toMatchSnapshot();
});