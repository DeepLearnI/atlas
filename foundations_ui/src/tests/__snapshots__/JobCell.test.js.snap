// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Renders Job Cell 1`] = `
<p
  className="job-cell"
/>
`;
